utility/src/test/java/org/oppia/android/util/profile/NameValidatorTest.kt

```diff
@@ -0,0 +1,26 @@
+package org.oppia.android.util.profile
+
+import com.google.common.truth.Truth
+import org.junit.Test
+
+class NameValidatorTest {
+
+  private val allowedNamesList = listOf<String>("जिष्णु", "Ben-Henning", "Rajat.T", "جيشنو")
+
+  private val unAllowedNamesList =
+    listOf<String>("जिष्णु7", "Ben_Henning", "Rajat..T", "جيشنو^&&", " ", ".", "Ben Henning")
+
+  @Test
+  fun testNameValidator_addUnAllowedName_returnFalse() {
+    unAllowedNamesList.forEach {
+      Truth.assertThat(!NameValidator.isNameValid(it))
+    }
```

Comment:
Each of the names should be a distinct test for each case being tested (such as English characters, Hindi characters, Portuguese characters, etc., allowed symbols, and failing cases such as disallowed symbols or characters).

Reply:
> English characters, Hindi characters, Portuguese characters
Could you please clarify which all language characters should I add tests for?

Reply:
I think the latest tests are sufficient.